runtime/lua/vim/shared.lua

```diff
@@ -1139,4 +1139,76 @@ function vim._defer_require(root, mod)
   })
 end
 
+--- @nodoc
+--- @class vim.context.mods
+--- @field buf? integer
+--- @field hide? boolean
+--- @field horizontal? boolean
+--- @field keepalt? boolean
+--- @field keepjumps? boolean
+--- @field keepmarks? boolean
+--- @field keeppatterns? boolean
+--- @field lockmarks? boolean
+--- @field noautocmd? boolean
+--- @field options? table<string, any>
+--- @field sandbox? boolean
+--- @field win? integer
+
+--- Executes function `f` with the given context specification.
+---
+--- @param context vim.context.mods
+function vim._with(context, f, ...)
```

Comment:
Was given a suggestion to end the function in varargs for the arguments for `f`, similarly to `pcall`, aka

```
vim._with(context, f, a, b)
```

would return the value of `f(a,b)`. Another suggestion if we wanted to avoid varargs would be to add an optional `args` field to `context` that could contain the arguments. I think both sound good for now, and went with the varargs options for now just to pick something. Since it's private for now we can afford to experiment/dogfood and update it.

Reply:
> suggestion to end the function in varargs for the arguments for `f`, similarly to `pcall`

that pcall "feature" makes it hard add overloads in the future. I think we should avoid that for now. It's a pretty low-value feature that really restricts flexibility in the future.

Reply:
not sure about the internals
but do we need this at all?
couldn't you just do `vim._with(context, function() return f(a,b))`?

Reply:
> but still I would wait on it. what's wrong with passing a closure?

Nah, nothing wrong with it. I think it was mostly a nicety thing in case the provided function was external. I'll revert it for now; if we notice we need it we can always introduce it again.

Reply:
> not sure about the internals
> but do we need this at all?
> couldn't you just do `vim._with(context, function() return f(a,b))`?

Indeed, this works. At the same time, `vim._with` was written with performance in mind and creating closure can be costly when run too frequently. In which case extra varargs are better (beside nicer looking calls). Though, this design matters only in some extreme and rare cases, so not a huge loss if it is not present.

Reply:
I think the performance of a closure is really minimal
and imo it's much less readable when using varargs

Reply:
> I think the performance of a closure is really minimal

As was said in previous comment: "... matters only in some extreme and rare cases". I certainly once had an instance when creating closures was a bottleneck. And it indeed was in a very performance heavy loop (around 300000 times expected to run without visible delay).

It only mattered if the overall design is deemed to be acceptable. Otherwise, not a huge loss.